tsa/tsacmd/server.go

```diff
@@ -83,22 +83,57 @@ func (forward ForwardedTCPIP) Wait() {
 	forward.Logger.Debug("drained")
 }
 
-func (server *server) Serve(listener net.Listener) {
+func (server *server) Serve(listener net.Listener) error {
+	var tempDelay time.Duration
 	for {
-		c, err := listener.Accept()
+		conn, err := listener.Accept()
 		if err != nil {
-			if !strings.Contains(err.Error(), "use of closed network connection") {
-				server.logger.Error("failed-to-accept", err)
+			if strings.Contains(err.Error(), "use of closed network connection") {
+				// shutting down
+				return nil
 			}
 
-			return
+			if ne, ok := err.(net.Error); ok && ne.Temporary() {
```

Comment:
That's interesting, I didn't know what that had to do with the comment around the number of open files

```go
func (e Errno) Temporary() bool {
	return e == EINTR || e == EMFILE || e.Timeout()
}

//        EMFILE The per-process limit on the number of open file descriptors has been reached.
```

([syscall_unix.go](https://github.com/golang/go/blob/ccd9d9d4cef16618d84bae9cba9f1b264278c8ab/src/syscall/syscall_unix.go#L123-L125))

Neat!